contracts/transfer/src/state.rs

```diff
@@ -190,9 +190,9 @@ impl TransferState {
             .extend_notes(block_height, tx_skeleton.outputs.clone());
 
         // if present, place the contract deposit on the state
-        if tx.payload().tx_skeleton().deposit > 0 {
-            let contract = match tx.payload().contract_call() {
-                Some(call) => ContractId::from_bytes(call.contract),
+        if tx.payload().tx_skeleton.deposit > 0 {
+            let contract = match &tx.payload().contract_call {
+                Some(call) => ContractId::from_bytes(call.contract.clone()),
```

Comment:
I think you don't even need to clone here

Reply:
Fixed